lib/src/kbaserelationengine/search/ElasticIndexingStorage.java

```diff
@@ -788,14 +811,115 @@ public void deleteAllVersions(final GUID guid) throws IOException {
     
     @Override
     public void publishObjects(Set<GUID> guids) throws IOException {
-        shareObjects(guids, PUBLIC_ACCESS_GROUP);
+        shareObjects(guids, PUBLIC_ACCESS_GROUP, false);
     }
     
     @Override
     public void unpublishObjects(Set<GUID> guids) throws IOException {
         unshareObjects(guids, PUBLIC_ACCESS_GROUP);
     }
 
+    @SuppressWarnings({ "serial", "unchecked" })
+    private boolean addExtPubForVersion(String indexName, GUID guid, 
+            int accessGroupId) throws IOException {
+        if (indexName == null) {
+            indexName = getAnyIndexPattern();
+        }
+        // Check that we work with other than physical access group this object exists in.
+        String pguid = new GUID(guid.getStorageCode(), guid.getAccessGroupId(),
+                guid.getAccessGroupObjectId(), guid.getVersion(), null, null).toString();
+        Map<String, Object> bool = new LinkedHashMap<String, Object>() {{
+            put("must", Arrays.asList(createFilter("term", "pguid", pguid)));
+        }};
+        Map<String, Object> query = new LinkedHashMap<String, Object>() {{
+            put("bool", bool);
+        }};
+        final HashMap<String, Object> params = new HashMap<>();
+        params.put("accgrp", accessGroupId);
+        Map<String, Object> script = new LinkedHashMap<String, Object>() {{
+            put("inline", "" + 
+                    "if (ctx._source.extpub.indexOf(params.accgrp) < 0) {\n" + 
+                    "  ctx._source.extpub.add(params.accgrp);\n" +
+                    "}\n");
+            put("params", params);
+        }};
+        Map<String, Object> doc = new LinkedHashMap<String, Object>() {{
+            put("query", query);
+            put("script", script);
+        }};
+        String urlPath = "/" + indexName + "/" + getAccessTableName() + "/_update_by_query";
+        Response resp = makeRequest("POST", urlPath, doc);
+        Map<String, Object> data = UObject.getMapper().readValue(
+                resp.getEntity().getContent(), Map.class);
+        return (Integer)data.get("updated") > 0;
+    }
+
+    @Override
+    public void publishObjectsExternally(Set<GUID> guids, int accessGroupId) throws IOException {
```

Comment:
This and unpublish aren't actually used, are they? They still need to be wired up to the event handler?

Reply:
They should but it's more tricky and I suggest to do it in separate PR. What do you think?

Reply:
Sure